model/src/main/kotlin/Identifier.kt

```diff
@@ -117,6 +118,12 @@ data class Identifier(
         return nameMatches && versionMatches
     }
 
+    /**
+     * Create a path based on the properties of the [Identifier]. All properties are encoded using [encodeOrUnknown].
+     */
+    fun toPath() = "${provider.encodeOrUnknown()}/${namespace.encodeOrUnknown()}/${name.encodeOrUnknown()}/" +
```

Comment:
Maybe some Kotlin fun?

    fun toPath() = listOf(provider, namespace, name, version).joinToString("/") { it.encodeOrUnknown() }

Reply:
Done.